include/gridtools/stencil_composition/sid/block.hpp

```diff
@@ -0,0 +1,82 @@
+/*
+ * GridTools
+ *
+ * Copyright (c) 2014-2019, ETH Zurich
+ * All rights reserved.
+ *
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+#pragma once
+
+#include "../../common/generic_metafunctions/for_each.hpp"
+#include "../../common/host_device.hpp"
+#include "../../common/hymap.hpp"
+#include "../../common/tuple_util.hpp"
+#include "../../meta/concat.hpp"
+#include "../../meta/push_back.hpp"
+#include "../../meta/rename.hpp"
+#include "../../meta/st_contains.hpp"
+#include "./concept.hpp"
+#include "./synthetic.hpp"
+
+namespace gridtools {
+    namespace sid {
+        template <class Dim>
+        struct blocked_dim {
+            using type = blocked_dim<Dim>;
+        };
+
+        namespace block_impl_ {
+            template <class Stride, class BlockSize>
+            struct blocked_stride {
+                Stride m_stride;
+                BlockSize m_block_size;
+            };
+
+            template <class Ptr, class Stride, class BlockSize, class Offset>
+            GT_FUNCTION auto sid_shift(Ptr &ptr, blocked_stride<Stride, BlockSize> const &stride, Offset const &offset)
+                GT_AUTO_RETURN(shift(ptr, stride.m_stride, stride.m_block_size *offset));
+
+            template <class Strides, class BlockMap>
+            struct block_strides_f {
+                Strides m_strides;
+                BlockMap m_map;
+
+                template <class Map, class Dim>
+                using decay_at = decay_t<decltype(at_key<Dim>(std::declval<Map>()))>;
+
+                template <class Dim>
+                blocked_stride<decay_at<Strides, Dim>, decay_at<BlockMap, Dim>> operator()(blocked_dim<Dim>) const {
+                    return {at_key<Dim>(m_strides), at_key<Dim>(m_map)};
```

Comment:
this will cause compilation error if blocking is requested on non existing in the original sid dimension